installer/lib/mix/tasks/phx.new.ex

```diff
@@ -329,7 +338,22 @@ defmodule Mix.Tasks.Phx.New do
   end
 
   defp check_app_name!(name, from_app_flag) do
-    unless name =~ Regex.recompile!(~r/^[a-z][a-z0-9_]*$/) do
+    with :ok <- validate_not_reserved(name),
+         :ok <- validate_app_name_format(name, from_app_flag) do
+      :ok
+    end
+  end
+
+  defp validate_not_reserved(name) when name in @reserved_app_names do
+    Mix.raise("Application name cannot be '#{name}' as it is reserved")
```

Comment:
For consistency with the other messages:

```suggestion
    Mix.raise("Application name cannot be #{inspect(name)} as it is reserved")
```